src/main/scala/beam/replanning/SupplementaryTripGenerator.scala

```diff
@@ -160,12 +160,9 @@ class SupplementaryTripGenerator(
             false -> noTrip,
           )
 
-        val makeTrip: Boolean = tripMNL.sampleAlternative(tripChoice, r).get.alternativeType
-
-        if (makeTrip) {
-          destinationMNL.sampleAlternative(modeChoice, r)
-        } else {
-          None
+        tripMNL.sampleAlternative(tripChoice, r) match {
+          case Some(mnlSample) if mnlSample.alternativeType => destinationMNL.sampleAlternative(modeChoice, r)
+          case _                                            => None
```

Comment:
How about a collect?

```suggestion
        tripMNL.sampleAlternative(tripChoice, r) collect {
          case mnlSample if mnlSample.alternativeType => destinationMNL.sampleAlternative(modeChoice, r)
```

Reply:
Sorry, I didn't pay attention to the final type - change my collect to a map and that would have been all that was needed.